.github/workflows/nightly.yml

```diff
@@ -0,0 +1,169 @@
+name: Nightly Build
+
+on:
+  # schedule:
+  #   # Run job at 10.30pm PST or 11.30pm PDT
+  #   - cron:  '30 6 * * *'
+  # FOR TESTING ONLY - gh workflow run nightly.yml --ref <branch_name>:
+  workflow_dispatch:
+
+jobs:
+  create-nightly-tag:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    permissions:
+      # Additional permission needed to generate tag
+      contents: write
+
+    outputs:
+      TAG: ${{ steps.create_tag.outputs.tag }}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Run make develop
+        run: make develop
+      - name: Create tag
+        id: create_tag
+        run: |
+          git config --global user.email "core+streamlitbot-github@streamlit.io"
+          git config --global user.name "Streamlit Bot"
+
+          TAG="$(./scripts/pypi_nightly_create_tag.py)"
+          echo "::set-output name=tag::$TAG"
+
+          ./scripts/update_version.py $TAG
+          ./scripts/update_name.py streamlit-nightly
+
+          git add lib/setup.py
+          git add frontend/package.json
+
+          git add lib/streamlit/__init__.py
+          git add lib/streamlit/version.py
+
+          git commit -m "Update version and project name in files"
+
+          git tag -a $TAG -m "Streamlit nightly $TAG"
+          git push origin $TAG
+      # TODO: Using testing webhook, switch to actual st-open-source channel webhook later
+      - if: ${{ failure() }}
+        name: Nightly Tag Failure Slack Message
+        env: 
+          SLACK_WEBHOOK: ${{ secrets.TEST_SLACK_WEBHOOK }}
+        run: |
+          curl -X POST -H 'Content-type: application/json' --data '{"text":":blobonfire: Nightly job failed to create a tag"}' "$SLACK_WEBHOOK"
+
+  run-python-tests:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/python-versions.yml
+    with:
+      ref: ${{needs.create-nightly-tag.outputs.tag}}
+
+  run-javascript-tests:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/js-tests.yml
+    with:
+      ref: ${{needs.create-nightly-tag.outputs.tag}}
+
+  run-py-prod-deps-smoke-test:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/py-prod-deps-smoke-test.yml
+    with:
+      ref: ${{needs.create-nightly-tag.outputs.tag}}
+
+  run-cypress-tests:
+    needs: create-nightly-tag
+    uses: ./.github/workflows/cypress.yml
+    with:
+      ref: ${{needs.create-nightly-tag.outputs.tag}}
+
+  test-status-notification:
+    runs-on: ubuntu-latest
+
+    if: ${{ always() }}
```

Comment:
It can't be dropped because by default, a job that has a `needs:` list only runs if all the needs jobs are successful. In this case we want it to always run, even if one or more of the test jobs fails.

Reply:
Ah, yeah that makes a lot more sense now.

In that case I'd just recommend adding a comment explaining why this is needed (which would also explain why the actual nightly job is gated by tests passing).